plugin/pkg/global/deletionconfirmation/admission.go

```diff
@@ -119,24 +129,70 @@ func (d *DeletionConfirmation) Admit(a admission.Attributes) error {
 		return admission.NewForbidden(a, errors.New("not yet ready to handle request"))
 	}
 
-	var obj metav1.Object
+	var (
+		obj         metav1.Object
+		cacheLookup func() (metav1.Object, error)
+		liveLookup  func() (metav1.Object, error)
+	)
 
 	switch a.GetKind().GroupKind() {
 	case garden.Kind("Shoot"):
-		shoot, err := d.shootLister.Shoots(a.GetNamespace()).Get(a.GetName())
-		if err != nil {
-			return err
+		cacheLookup = func() (metav1.Object, error) {
+			shoot, err := d.shootLister.Shoots(a.GetNamespace()).Get(a.GetName())
+			if err != nil {
```

Comment:
why not just immediately return the value of the function (the 'cast' should happen implicitly)